### PR TITLE
Add Apex Block Storage for AWS

### PR DESCRIPTION
**Pull request by shekhar-j**
Add Apex block storage for AWS. 

Destroy complete! Resources: 36 destroyed

Screenshot 2024-11-30 122619

### CODE REVIEW

File: modules/aws_infra/main.tf [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# */
+
+#resource "aws_vpc" "apex_vpc" {
```
Comment (doriac11) [inline modules/aws_infra/main.tf:18]: Should this be deleted?

**Review by doriac11 — APPROVED** [review]

**Review by rfsonders — COMMENTED** [review]
I have tested multiple deployments with the various permutations of configurations. Balanced, Performance, Multi-AZ. All are working as required.

**Comment by zhangsan946** [comment]
How does it support different package bundle version? I didn't see a param for it.

**Comment by vangork** [comment]
How does it support different package bundle version? I didn't see a param for it.

**Comment by shekhar-j** [comment]
> How does it support different package bundle version? I didn't see a param for it.

Either application version or ami-id will pickup specified version image.

File: examples/aws_install/main.tf [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+module "aws_install" {
+  source = "../../modules/aws_install"
+
+  installer_node_ip = module.aws_infra.installer_ip
+  co_res_ips = module.aws_infra.co_res_ips
+  device_mapping = module.aws_infra.device_mapping
+  loadbalancer_dns = module.load-balancer.loadbalancer_dns
+  loadbalancer_ip = module.load-balancer.loadbalancer_private_ip
+  #node_hostnames = var.node_hostnames
```
Comment (IsaiasA1) [inline examples/aws_install/main.tf:60]: Is this necessary to keep line 60 commented out? or have at all if it's not used?

**Review by IsaiasA1 — APPROVED** [review]

**Pull request merged**